### DISCUSSION TITLE
[BUG] - Tandoor migration does not parse ingredients/units if activated

### DISCUSSION THREAD

**Discussion by lmgarret** [mealie-recipes/mealie#2529] [Feature Request]
### First Check

- [X] This is not a feature request
- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.
- [X] This issue can be replicated on the demo site (https://demo.mealie.io/)

### What is the issue you are experiencing?

I'm not sure if this is a bug or a missing feature, I went with the former. The group I'm in has 'disable ingredient and units parsing/organization' toggled off. I imported some recipes from Tandoor, with ingredients and units set. The resulting recipes have a `1` as a quantity and no ingredient/unit set

### Steps to Reproduce

1. Set the group option 'disable ingredient and units parsing/organization' to OFF
2. Import recipes from tandoor using the migration tool. The recipes should have 'rich' units and ingredients. Sample export provided
3. Check in the imported recipe the ingredients quantities, they're all set to 1.
[tandoor_sample.zip](https://github.com/mealie-recipes/mealie/files/12466295/tandoor_sample.zip)

### Please provide relevent logs

No logs 

### Mealie Version

Latest nightly, using omni image. docker image sha256: 8ec73b5781646d77f6e0851b570de741c3d5dba38a12f364568076ebc12cba39

### Deployment

Docker (Linux)

**Comment by michael-genson** [comment]
Currently we don't support migrating Tandoor's ingredients and units out of the box. You'll need to parse them yourself for now. 

There are a few reasons for this (including the fact that [the Tandoor migration is fairly new](https://github.com/mealie-recipes/mealie/pull/2438)), but mainly it has to do with how to handle matching foods/units to the existing food/unit store, as well as how to handle missing foods/units. Right now we don't have a reliable way to do the matching, so we either need to:
1. ignore the existing food store and blindly create new foods/units defined in Tandoor, probably creating duplicates
2. wipe the group's existing foods/units to ensure there are no duplicates, then create new ones using the Tandoor definitions

This is only a problem if you already have foods/units defined, which is unlikely since you're probably only migrating when you first spin up Mealie, however it _also_ means that you can't really use the seed function anymore (since, for the same reasons, it does not match to existing foods or prevent duplicates), which many users find super useful (I know I did when I started using Mealie).

I'd be open to working on a feature request to add support for conserving Tandoor's foods and units when migrating, but I'm hesitant until we refine food/unit matching. I could see a world where we add the option with a warning explaining the caveats above though, even before the matching issue is resolved. Any thoughts on that?

**Reply by lmgarret** [reply]
Thanks for the clear explanation! I understand the challenges that it would bring, especially if the units/ingredients have already been populated.. I assume the behavior is the same for all import types and not only Tandoor?

In the case of an empty database, I'd argue importing ingredients from the export may sometimes be more useful than the seeding feature. The seeding feature is really great to get started, but if someone already has a large collection of recipes, chances are there will be more ingredients in the collection than in the seeding database? (to be honest I don't remember how many ingredients there are in the seeding db). But this is data dependent so we cannot take a decision for the user here of course.

I like your suggestion of an option on the migration screen. It would fit the use case of migrating a large collection of recipes with all ingredients/units properly setup. (Until the matching has been refined as you said)
